Sources/SwiftProtobuf/JSONEncoder.swift

```diff
@@ -148,13 +173,19 @@ internal struct JSONEncoder {
                 append(staticText: "\"Infinity\"")
             }
         } else {
-            // TODO: Be smarter here about choosing significant digits
-            // See: protoc source has C++ code for this with interesting ideas
             if let v = Int64(safely: value) {
```

Comment:
If we skip this and just do the lower logic, what gets printed for whole number? Seemed like the C++ code just used the format string no matter what.